test/source/browser/browser-pool.ts

```diff
@@ -148,7 +110,7 @@ export class BrowserPool {
       try {
         const browser = await withTimeouts(this.newBrowserHandle(t));
         try {
-          await withTimeouts(this.cbWithTimeout(async () => await cb(t, browser), consts.TIMEOUT_EACH_RETRY));
+          await withTimeouts(this.cbWithTimeout(async () => cb(t, browser), consts.TIMEOUT_EACH_RETRY));
```

Comment:
I suppose the `cb` returns a promise? If yes, it may seem like there is no point in awaiting the promise right here, and can just return the promise, and let the higher level function await it. But that's not entirely the same. By awaiting it explicitly, and always, we ensure that the stack trace will be complete. By passing the `Promise` around without awaiting it, we may lose out on some parts of the stack trace. That was the case on Node 12 and browsers of a few years ago, anyway. The situation I think somewhat improved, but I also think there are probably still gaps, so I generally recommend to never return a `Promise` without first awaiting it.

The function signature of course remains the same either way, it's just whether or not to include the `await` - which I strongly recommend to include.

Reply:
`cb` here returns `void`:
```ts
public withNewBrowserTimeoutAndRetry = async (cb: (t: AvaContext, browser: BrowserHandle) => void, ...
```

Also vscode notified that `await` isn't needed:
<img width="500" alt="Screenshot 2023-02-01 at 15 09 06" src="https://user-images.githubusercontent.com/1062093/216052251-bb265913-495e-40a3-9f82-8a23bccffb34.png">

Reply:
Yup, if it returns void, then my comment is also void.